### PR TITLE
node-proxy: make the daemon discover endpoints

### PR DESCRIPTION
**Pull request by guits**
Add logic in order to explore the API.
This will allow node-proxy to be compatible with more
hardware.

Fixes: https://tracker.ceph.com/issues/65394

Signed-off-by: Guillaume Abrioux <gabrioux@ibm.com>

### CODE REVIEW

**Comment by guits** [comment]
https://pulpito.ceph.com/gabrioux-2024-04-29_12:40:30-orch:cephadm-wip-guits-testing-3-2024-04-10-1153-distro-default-smithi/

File: src/pybind/mgr/cephadm/inventory.py [hunk 1]
```diff
@@ -1500,19 +1500,29 @@ def is_error(statuses: ValuesView) -> bool:
...
-            for component in data['status'].keys():
-                values = data['status'][component].values()
-                if is_error(values):
-                    state = 'error'
-                elif is_unknown(values):
+            for component, details in data['status'].items():
+                _sys_id_res: List[str] = []
+                for element in details.values():
+                    values = element.values()
+                    if is_error(values):
```
Comment (adk3798) [inline src/pybind/mgr/cephadm/inventory.py:1509]: some unit tests are failing
```
FAILED cephadm/tests/test_node_proxy.py::TestNodeProxyEndpoint::test_data_raises_alert
FAILED cephadm/tests/test_node_proxy.py::TestNodeProxyEndpoint::test_summary_no_hostname
FAILED cephadm/tests/test_node_proxy.py::TestNodeProxyEndpoint::test_summary_with_valid_hostname
```
and it seems to be coming from this call
```
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/.tox/py3/lib/python3.10/site-packages/cherrypy/_cprequest.py", line 638, in respond
    self._do_respond(path_info)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/.tox/py3/lib/python3.10/site-packages/cherrypy/_cprequest.py", line 697, in _do_respond
    response.body = self.handler()
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/.tox/py3/lib/python3.10/site-packages/cherrypy/lib/encoding.py", line 223, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/.tox/py3/lib/python3.10/site-packages/cherrypy/lib/jsontools.py", line 59, in json_handler
    value = cherrypy.serving.request._json_inner_handler(*args, **kwargs)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/.tox/py3/lib/python3.10/site-packages/cherrypy/_cpdispatch.py", line 54, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/cephadm/agent.py", line 447, in summary
    results = self.mgr.node_proxy_cache.summary(**kw)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/cephadm/inventory.py", line 1509, in summary
    if is_error(values):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/cephadm/inventory.py", line 1496, in is_error
    return any([status['status']['health'].lower() == 'error' for status in statuses])
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/mgr/cephadm/inventory.py", line 1496, in <listcomp>
    return any([status['status']['health'].lower() == 'error' for status in statuses])
TypeError: string indices must be integers
```

**Comment by guits** [comment]
https://pulpito.ceph.com/gabrioux-2024-05-14_05:57:55-orch:cephadm-wip-guits-testing-2024-05-13-1110-distro-default-smithi/

**Review by asm0deuz — APPROVED** [review]
LGTM

**Pull request merged**